sherpa/plot/bokeh_backend.py

```diff
@@ -0,0 +1,829 @@
+        x1 : array-like or scalar number
+            upper bin boundary values
+        y : array-like or scalar number
+            y values, same dimension as `x0`.
+        {kwargs}
+        """
+        if linecolor is not None:
+            logger.warning("The linecolor attribute ({}) is unused.".format(linecolor))
+
+        x, y2 = histogram_line(xlo, xhi, y)
+        # Note: this handles clearing the plot if needed.
+        #
+        objs = self.plot(x, y2, yerr=None, xerr=None,
+                title=title, xlabel=xlabel, ylabel=ylabel,
+                overplot=overplot, clearwindow=clearwindow,
+                xerrorbars=False, yerrorbars=False,
+                ecolor=ecolor, capsize=capsize,
+                xlog=xlog, ylog=ylog,
+                linestyle=linestyle,
+                linewidth=linewidth,
+                drawstyle=drawstyle,
+                color=color, marker=None, alpha=alpha,
+                xaxis=False, ratioline=False)
+
+        # Draw points and error bars at the mid-point of the bins.
+        # Use the color used for the data plot: should it
+        # also be used for marker[face]color and ecolor?
+        #
+        xmid = 0.5 * (xlo + xhi)
+        xerr = (xhi - xlo) / 2 if xerrorbars else None
+        yerr = yerr if yerrorbars else None
+
+        try:
+            color = objs[0].get_color()
+        except (AttributeError, IndexError):
+            pass
+
+        # Do not draw a line connecting the points
+        #
+        # Unlike plot, using errorbar for both cases.
+        #
+
+        ebars = self.plot(xmid, y,
+                          # Plot error y bars if present
+                          xerrorbars=False,
+                          yerrorbars=True,
+                          yerr=yerr, xerr=xerr,
+                          overplot=True,
+                          clearwindow=False,
+                          color=color,
+                          alpha=alpha,
+                          linestyle='noline',
+                          marker=marker,
+                          markersize=markersize,
+                          markerfacecolor=markerfacecolor,
+                          ecolor=ecolor,
+                          capsize=capsize,
+                         )
+
+    # Note that this is an internal method that is not wrapped in
+    # @ translate_args
+    # This is called from methods like plot, histo, etc.
+    # so the arguments passed to this method are already translated and we do
+    # not want to apply the translation a second time.
+    def _set_line(self, line, linecolor=None, linestyle=None, linewidth=None):
+        """Apply the line attributes, if set.
+
+        Parameters
+        ----------
+        line :
+            The line to change
+        linecolor, linestyle, linewidth : optional
+            The attribute value or None.
+        """
+
+        def set(label, newval):
+            func = getattr(line, 'set_' + label)
+            func(newval)
```

Comment:
```suggestion
```

I don't think we use `set` here, unlike in matplotlib (or the old matplotlib code,I haven't checked recently).